src/Nethermind/Nethermind.Blockchain.Test/TxPools/Collections/SortedPoolTests.cs

```diff
@@ -31,13 +35,20 @@ public class SortedPoolTests
     {
         private const int Capacity = 16;
 
-        private readonly SortedPool<Keccak, Transaction, Address> _sortedPool = new TxSortedPool(Capacity);
+        private SortedPool<Keccak, Transaction, Address> _sortedPool;
 
         private Transaction[] _transactions = new Transaction[Capacity * 8];
         
         [SetUp]
         public void Setup()
         {
+            ISpecProvider specProvider = Substitute.For<ISpecProvider>();
+            IBlockTree blockTree = Substitute.For<IBlockTree>();
+            Block block =  Build.A.Block.WithNumber(0).TestObject;
+            blockTree.Head.Returns(block);
+            specProvider.GetSpec(Arg.Any<long>()).Returns(new ReleaseSpec() {IsEip1559Enabled = false});
+            ITransactionComparerProvider transactionComparerProvider = new TransactionComparerProvider(specProvider, blockTree);
```

Comment:
confusing -> what does GetDefaultComparer do?

Reply:
why not simple new TxComparer(specProvider, blockTree) here
it seems that we create a provider, use it once and then discard?